example_glue.py

```diff
@@ -0,0 +1,525 @@
+import gobasis
+import grid
+import iodata
+import numpy as np
+from functools import wraps
+from meanfield import *
+
+
+#
+# Decorators
+#
+
+def onetime(varname):
+    def decorator(func):
+        @wraps(func)
+        def wrapper(self, *args, **kwargs):
+            if getattr(self, varname) is not None:
+                print("Trying to set a one-time attribute {varname}. Ignored")
+                return
+            func(self, *args, **kwargs)
+
+        return wrapper
+
+    return decorator
+
+
+def cache(varname):
+    def decorator(func):
+        @wraps(func)
+        def wrapper(self, *args, **kwargs):
+            if getattr(self, varname) is None:
```

Comment:
`getattr` will raise an exception if the attribute is not present and when no default argument is provided. Would be better to have one `getattr` call also. E.g. like

```python
val = getattr(self, varname, None)
if val is None:
    val = func(self, *args, **kwargs)
    setattr(self, varname, val)
return val
```